lib/cisco_node_utils/vxlan_global.rb

```diff
@@ -0,0 +1,158 @@
+    rescue Cisco::CliError => e
+      # This cmd will syntax reject if feature is not
+      # enabled. Just catch the reject and return false.
+      return false if e.clierror =~ /Syntax error/
+      raise
+    end
+
+    # ----------
+    # PROPERTIES
+    # ----------
+
+    # dup-host-ip-addr-detection
+    def dup_host_ip_addr_detection
+      get_args = {}
+      match = config_get('vxlan_global', 'dup_host_ip_addr_detection', get_args)
+      if match.nil? || match.first.nil?
+        default_dup_host_ip_addr_detection
+      else
+        match.first.collect(&:to_i)
+      end
+    end
+
+    def dup_host_ip_addr_detection_host_moves
+      host_moves, _timeout = dup_host_ip_addr_detection
+      return default_dup_host_ip_addr_detection_host_moves if host_moves.nil?
+      host_moves
+    end
+
+    def dup_host_ip_addr_detection_timeout
+      _host_moves, timeout = dup_host_ip_addr_detection
+      return default_dup_host_ip_addr_detection_timeout if timeout.nil?
+      timeout
+    end
+
+    def dup_host_ip_addr_detection_set(state, host_moves, timeout)
+      set_args = {}
+      set_args[:state] = state
+      set_args[:host_moves] = host_moves
+      set_args[:timeout] = timeout
+
+      config_set('vxlan_global', 'dup_host_ip_addr_detection', set_args)
+    end
+
+    def default_dup_host_ip_addr_detection
+      [default_dup_host_ip_addr_detection_host_moves,
+       default_dup_host_ip_addr_detection_timeout]
+    end
+
+    def default_dup_host_ip_addr_detection_host_moves
+      config_get_default('vxlan_global',
+                         'dup_host_ip_addr_detection_host_moves')
+    end
+
+    def default_dup_host_ip_addr_detection_timeout
+      config_get_default('vxlan_global', 'dup_host_ip_addr_detection_timeout')
+    end
+
+    # dup-host-mac-detection
+    def dup_host_mac_detection
+      get_args = {}
+      match = config_get('vxlan_global', 'dup_host_mac_detection', get_args)
+      if match.nil? || match.first.nil?
+        default_dup_host_mac_detection
+      else
+        match.first.collect(&:to_i)
+      end
+    end
+
+    def dup_host_mac_detection_host_moves
+      host_moves, _timeout = dup_host_mac_detection
+      return default_dup_host_mac_detection_host_moves if host_moves.nil?
+      host_moves
+    end
+
+    def dup_host_mac_detection_timeout
+      _host_moves, timeout = dup_host_mac_detection
+      return default_dup_host_mac_detection_timeout if timeout.nil?
+      timeout
+    end
+
+    def dup_host_mac_detection_set(host_moves, timeout)
+      set_args = {}
+      set_args[:host_moves] = host_moves
+      set_args[:timeout] = timeout
+
+      config_set('vxlan_global', 'dup_host_mac_detection', set_args)
+    end
+
+    def dup_host_mac_detection_default
+      config_set('vxlan_global', 'dup_host_mac_detection_default')
+    end
+
+    def default_dup_host_mac_detection
+      [default_dup_host_mac_detection_host_moves,
+       default_dup_host_mac_detection_timeout]
```

Comment:
I'd prefer to see this in the yaml file instead, as:


  dup_host_ip_addr_detection:
    ...
    ...
    ...
    default: [default_dup_host_mac_detection_host_moves, default_dup_host_mac_detection_timeout]